pkg/controller/core/workload_controller.go

```diff
@@ -158,18 +159,39 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	log := ctrl.LoggerFrom(ctx)
+	status := workload.Status(&wl)
+	log := ctrl.LoggerFrom(ctx).WithValues("workload", klog.KObj(&wl), "queue", wl.Spec.QueueName, "status", status)
 	log.V(2).Info("Reconcile Workload")
 
-	if len(wl.OwnerReferences) == 0 && !wl.DeletionTimestamp.IsZero() {
-		// manual deletion triggered by the user
-		err := workload.RemoveFinalizer(ctx, r.client, &wl)
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+	if !wl.DeletionTimestamp.IsZero() {
+		log = log.WithValues("deletionTimestamp", wl.DeletionTimestamp)
+		log.Info("Attemtping to finalize workload.")
+
+		switch {
+		case controllerutil.ContainsFinalizer(&wl, kueue.ResourceInUseFinalizerName):
+			{
+				log.Info("Manual deletion by a user detected.")
+				if len(wl.OwnerReferences) == 0 {
+					return ctrl.Result{}, r.finalize(ctx, &wl, log)
+				} else {
+					log.Info("Uable to finalize: workload still has owners. Proceeding with reconcile.", "owners", wl.OwnerReferences)
+				}
+			}
+		case controllerutil.ContainsFinalizer(&wl, kueue.SafeDeleteFinalizerName):
+			{
+				return ctrl.Result{}, r.finalize(ctx, &wl, log)
+			}
+		default:
+			{
+				log.Info("Unknown finalizer(s) present. Proceeding with reconcile.")
+			}
+		}
 	}
 
 	finishedCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadFinished)
 	if finishedCond != nil && finishedCond.Status == metav1.ConditionTrue {
 		if !features.Enabled(features.ObjectRetentionPolicies) || r.workloadRetention == nil || r.workloadRetention.afterFinished == nil {
+			log.Info("Unable to determine workload retention scheme.")
```

Comment:
The logs in this function seem pretty verbose, shouldn't the level of those logs be higher? Here you are using `0`, other functions use `2` etc.

Reply:
Overlook on my part, correcting.